main/src/ca/uwaterloo/flix/api/lsp/LspServer.scala

```diff
@@ -304,6 +305,16 @@ object LspServer {
       CompletableFuture.completedFuture(references.map(_.toLsp4j).toList.asJava)
     }
 
+    override def rename(params: RenameParams): CompletableFuture[WorkspaceEdit] = {
+      val newName = params.getNewName
+      val uri = params.getTextDocument.getUri
+      val pos = Position.fromLsp4j(params.getPosition)
+      RenameProvider.processRename(newName, uri, pos)(flixLanguageServer.root) match {
+        case Some(rename) => CompletableFuture.completedFuture(rename.toLsp4j)
+        case None => CompletableFuture.completedFuture(new WorkspaceEdit())
```

Comment:
I would add that comment from the PR-- that if nothing is found its OK to return the empty workspace edit.